source/android/mobilechatapp/src/main/java/com/example/mobilechatapp/MainActivity.java

```diff
@@ -263,22 +263,32 @@ public void run()
         this.runOnUiThread(new RunnableExtended(this, text, duration));
     }
 
-    private void onSubmit(BaseActionElement actionElement, RenderedAdaptiveCard renderedAdaptiveCard) {
+    private void onSubmit(BaseActionElement actionElement, RenderedAdaptiveCard renderedAdaptiveCard)
+    {
         SubmitAction submitAction = null;
-        if (actionElement instanceof SubmitAction) {
+        if (actionElement instanceof SubmitAction)
+        {
             submitAction = (SubmitAction) actionElement;
-        } else if ((submitAction = SubmitAction.dynamic_cast(actionElement)) == null) {
+        }
+        else if ((submitAction = SubmitAction.dynamic_cast(actionElement)) == null)
+        {
             throw new InternalError("Unable to convert BaseActionElement to ShowCardAction object model.");
         }
 
         String data = submitAction.GetDataJson();
         Map<String, String> keyValueMap = renderedAdaptiveCard.getInputs();
         if (!data.isEmpty())
         {
-            try {
-                JSONObject object = new JSONObject(data);
+            try
+            {
+                JSONObject object = new JSONObject();
```

Comment:
> = new JSONObject(); [](start = 33, length = 20)

maybe do this in an `else` case below to avoid an unnecessary allocation? (also in the other file)